craft_archives/repo/package_repository.py

```diff
@@ -79,17 +92,19 @@ def unmarshal_package_repositories(cls, data: Any) -> List["PackageRepository"]:
 class PackageRepositoryAptPPA(PackageRepository):
     """A PPA package repository."""
 
-    def __init__(self, *, ppa: str) -> None:
+    def __init__(self, *, ppa: str, priority: Optional[int] = None) -> None:
         self.type = "apt"
         self.ppa = ppa
+        self.priority = priority
 
         self.validate()
 
     @overrides
-    def marshal(self) -> Dict[str, Any]:
+    def marshal(self) -> Dict[str, Union[str, int]]:
```

Comment:
Now I wonder why these classes don't use pydantic models right the rest of the craft-* code? Maybe we should look into that for the future

Reply:
Based on the copyright headers I think a lot of this predates pydantic. I've made #41 so we remember